File: java-hangman-app/src/test/java/ru/polina_project/hangman/util/ParserTest.java
package ru.polina_project.hangman.util;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.io.*;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;
import static ru.polina_project.hangman.util.Parser.loadWords;

public class ParserTest {
    private static Path tempFile;

    @BeforeEach
    public void setUp() throws IOException {
        tempFile = Files.createTempFile("test-words", ".txt");
        List<String> words = List.of("word1", "word2", "word3", "word4");
        Files.write(tempFile, words);
    }
    @AfterEach
    public void  tearDown() throws IOException {
        Files.deleteIfExists(tempFile);
    }

    @Test
    public void testLoadWords() {
        String filePath = tempFile.toString();
        List<String> expectedWords = List.of("word1", "word2", "word3", "word4");
        List<String> actualWords = Parser.loadWords(filePath);
        assertEquals(expectedWords, actualWords);

    }



//    @Test
//    public void setUp() throws IOException {
//        Path tempFile = Files.createTempFile("test-words", ".txt");
//        Files.write(tempFile, List.of("word1", "word2", "word3"));
//
//        List<String> words = Parser.loadWords(tempFile.toString());
//
//        assertNotNull(words);
//        assertEquals(3, words.size());
//        assertTrue(words.contains("word1"));
//        assertTrue(words.contains("word2"));
//        assertTrue(words.contains("word3"));
//
//        Files.deleteIfExists(tempFile);
//    }
}

File: kotlin-hangman-app/src/test/kotlin/ru/polina_project/hangman/util/ParserTest.kt
package ru.polina_project.hangman.util

import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Files.deleteIfExists
import java.nio.file.Files.write
import java.nio.file.Path

class ParserTest {

    var tempFile: Path? = null

    @BeforeEach
    @Throws(IOException::class)
    fun setUp() {
        tempFile = Files.createTempFile("test-words", ".txt")
        val words = listOf("word1", "word2", "word3", "word4")
        write(tempFile, words)
    }

    @AfterEach
    @Throws(IOException::class)
    fun tearDown() {
        deleteIfExists(tempFile)
    }

    @Test
    fun testLoadWords() {
        val filePath = tempFile.toString()
        val expectedWords = listOf("word1", "word2", "word3", "word4")
        val actualWords: List<String> =loadWords(filePath)
        assertEquals(expectedWords, actualWords)
    }
}